_posts/2023-02-22-user-defined-operators.md

```diff
@@ -0,0 +1,362 @@
+Member access via `.`, subscripting arrays with `[]`, or making a function call by adding parentheses
+to a function can be thought of as applications of operators as well.
+Even putting something in parentheses creates a valid operator expression.
+
+It is also important to note that in many languages assignment is a statement that is handled specially.
+In Solidity, like in many C-like languages it took inspiration from, assignment can be freely used in
+expressions like other operators.
+
+In most cases the built-in operators are only defined for [value types](https://docs.soliditylang.org/en/latest/types.html#value-types).
+Notable exceptions are the simple assignment and the "miscellaneous" group.
+For example, you can put almost anything in parentheses or branches of the ternary operator, as a part
+of a bigger expression, even if its type cannot be named or even assigned to anything.
+
+#### `using for`
+Since operators are a purely syntactic element, you can always use ordinary functions as a substitute.
+Solidity has a mechanism that is often used this way:
+[`using for`](https://docs.soliditylang.org/en/latest/contracts.html#using-for).
+
+The directive `using ... for ...` can be used to attach functions to a type.
+These functions can then be used as members on any object of that type and receive the object as
+their first parameter.
+
+```solidity
+type Int is int;
+using {add} for Int;
+
+function add(Int a, Int b) pure returns (Int) {
+    return Int.wrap(Int.unwrap(a) + Int.unwrap(b));
+}
+
+function test(Int a, Int b) pure returns (Int) {
+    return a.add(b);
+}
+```
+
+Currently, `using for` has 3 forms:
+1. `using L for T`: attaches all non-private functions from library `L` to type `T`.
+1. `using L for *`: attaches all non-private functions from library `L` to any type, including types that
+    cannot be named in the language and used explicitly (e.g. types of literals or array slices).
+1. `using {f, g, L.h} for T`: Attaches the given free or library functions `f`, `g`, and `L.h` to type `T`.
+
+The directive can be used:
+- ✅ At the file level.
+- ✅ Inside contract and library definitions.
+
+Can not be used:
+- ❌ Inside interface definitions.
+
+When used inside contracts, it is not inherited.
+Its effects extend only to the base contract that has it and not to any other contracts derived from it.
+
+At the file level, the directive affects all the free functions, contracts and libraries defined within that file.
+It does not affect other files that import the file containing it, unless it is marked as `global`.
+This extends its effect to all the files where the type is visible.
+The use of `global` is possible only with UDVTs, structs, and enums that are defined in the current source unit.
+
+### Introducing User-Defined Operators
+The new feature extends the third form of the `using for` syntax, so that the function, instead of being
+attached to the type, it is instead assigned an operator symbol.
```

Comment:
```suggestion
The new feature extends the third form of the `using for` syntax, so that the
function is assigned an operator symbol, instead of being attached to the type.
```